uint/src/uint.rs

```diff
@@ -1219,16 +1195,24 @@ macro_rules! impl_std_for_uint {
 					return write!(f, "0");
 				}
 
-				let mut s = String::new();
+				let mut buf = [0_u8; $n_words*20];
```

Comment:
Why `$n_words * 20`? Surely it's `ceil(log10(2^($n_words - 1)))`? That's 154 for U512 but this code would create a buffer of size 1280 (way too big).

Reply:
This change came from https://github.com/paritytech/bigint/blob/master/src/uint.rs#L927 (part of PR [#43](https://github.com/paritytech/bigint/pull/43)) – not sure what their thinking was there, maybe it was making space for a somewhat long debug string (which doesn't make sense).
Should I just go ahead and replace it?

Reply:
Yes please!

Reply:
Wait a sec, I think you're confusing `uint` with `fixed-hash`: here the `construct_uint!` macro is called with the **words** not the number of bytes, so `$n_words` here is `8` => `8 * 10 == 160` which is fine I think.

Reply:
The `construct_hash!` macro takes number of bytes, which is more logical imo. Quite confusing.